src/query/expression/src/types/array.rs

```diff
@@ -370,7 +370,12 @@ impl<T: ValueType> ArrayColumnBuilder<T> {
     pub fn append_column(&mut self, other: &ArrayColumn<T>) {
         // the first offset of other column may not be zero
         let other_start = *other.offsets.first().unwrap() as usize;
-        let other_end = *other.offsets.last().unwrap() as usize;
+        let mut other_end = *other.offsets.last().unwrap() as usize;
+
+        if other_end == 0 {
```

Comment:
We don't need to change the code here

Reply:
No worries, it's a temp fix. We can skip it.